src/dotnet/commands/dotnet-add/Program.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectJsonMigration;
+using NuGet.Frameworks;
+using Microsoft.DotNet.Tools.Add.ProjectToProjectReference;
+
+namespace Microsoft.DotNet.Tools.Add
+{
+    public class AddCommand
+    {
+        public const string HelpText = @".NET Add Command
+
+Usage: dotnet add [options] [object] <command> [[--] <arg>...]]
+
+Arguments:
+  <object>   The object of the operation. If a project file is not specified, it defaults to the current directory.
+  <command>  Command to be executed on <object>.
+
+Options:
+  -h|--help  Show help information
+
+Args:
+  Any extra arguments passed to the command.
+
+Commands:
+  p2p        Add project to project (p2p) reference to a project";
+
+        private static Dictionary<string, Func<string[], int>> s_builtIns = new Dictionary<string, Func<string[], int>>
+        {
+            ["p2p"] = AddProjectToProjectReferenceCommand.Run,
+        };
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            if (args.Length == 0 || args[0] == "--help" || args[0] == "-h")
```

Comment:
It is a pity that you can't use the CommandLineApplication here, because it needs to be used in the internal command.